### ISSUE TITLE
Constants are possibly incorrect

### ISSUE THREAD

**Issue by ghost** [adafruit/Adafruit_CircuitPython_INA219#5]
I think these are wrong:

```python
_CONFIG_BADCRES_MASK             = const(0x0780)  # Bus ADC Resolution Mask
_CONFIG_BADCRES_9BIT             = const(0x0080)  # 9-bit bus res = 0..511
_CONFIG_BADCRES_10BIT            = const(0x0100)  # 10-bit bus res = 0..1023
_CONFIG_BADCRES_11BIT            = const(0x0200)  # 11-bit bus res = 0..2047
_CONFIG_BADCRES_12BIT            = const(0x0400)  # 12-bit bus res = 0..4097
```

Based on Table 5 in the [IN219 datasheet](http://www.ti.com/lit/ds/symlink/ina219.pdf), they should probably be the following, inserting 0 for all X (don't-cares), and changing 4097 to 4095 in the comment.

```python
_CONFIG_BADCRES_MASK             = const(0x0780)  # Bus ADC Resolution Mask
_CONFIG_BADCRES_9BIT             = const(0x0000)  # 9-bit bus res = 0..511
_CONFIG_BADCRES_10BIT            = const(0x0080)  # 10-bit bus res = 0..1023
_CONFIG_BADCRES_11BIT            = const(0x0100)  # 11-bit bus res = 0..2047
_CONFIG_BADCRES_12BIT            = const(0x0180)  # 12-bit bus res = 0..4095
```

Also, for the "BADCRES" bits, you've left off the rest of Table 5, even though it is included for the "SADCRES" bits. The averaging features area available for both. Perhaps all those constants should be refactored to use offsets so they can be shared between BADCRES and SADCRES.

**Comment by paulswirhun** [comment]
I haven't tested this (awaiting assembled boards now), but I ended up rewriting this module to change the style, but based on your example. In particular, there is no need to be so verbose about setting a single config register, in my opinion.

```python
from adafruit_bus_device.i2c_device import I2CDevice
from micropython import const

# Registers
_REG_CONFIG                       = const(0x00)
_REG_SHUNTVOLTAGE                 = const(0x01)
_REG_BUSVOLTAGE                   = const(0x02)
_REG_POWER                        = const(0x03)
_REG_CURRENT                      = const(0x04)
_REG_CALIBRATION                  = const(0x05)

# Masks and offsets in the config register
_B_ADC_OFFSET                     = const(7)
_S_ADC_OFFSET                     = const(3)
_CONFIG_B_VOLTAGERANGE_MASK       = const(0x2000)  # Bus Voltage Range Mask
_CONFIG_GAIN_MASK                 = const(0x1800)  # Gain Mask
_CONFIG_B_ADC_MASK                = const(0x0780)
_CONFIG_S_ADC_MASK                = const(0x0078)
_CONFIG_MODE_MASK                 = const(0x0007)

# Bits
_CONFIG_RESET                     = const(0x8000)  # Reset by setting this bit.
_CONFIG_B_VOLTAGERANGE_16V        = const(0x0000)  # 0-16V Range
_CONFIG_B_VOLTAGERANGE_32V        = const(0x2000)  # 0-32V Range
_CONFIG_GAIN_1_40MV               = const(0x0000)  # Gain 1, 40mV Range
_CONFIG_GAIN_2_80MV               = const(0x0800)  # Gain 2, 80mV Range
_CONFIG_GAIN_4_160MV              = const(0x1000)  # Gain 4, 160mV Range
_CONFIG_GAIN_8_320MV              = const(0x1800)  # Gain 8, 320mV Range

# Table 5 in datasheet (ADC settings). Shift these left by _{B|S}_ADC_OFFSET
# to align them in the config register. Comment shows conversion time.
_ADC_SETTINGS_9BIT                = const(0x0)  # 84 us
_ADC_SETTINGS_10BIT               = const(0x1)  # 148 us
_ADC_SETTINGS_11BIT               = const(0x2)  # 276 us
_ADC_SETTINGS_12BIT               = const(0x3)  # 532 us
_ADC_SETTINGS_12BIT_AVG2          = const(0x9)  # 1.06 ms
_ADC_SETTINGS_12BIT_AVG4          = const(0xA)  # 2.13 ms
_ADC_SETTINGS_12BIT_AVG8          = const(0xB)  # 4.26 ms
_ADC_SETTINGS_12BIT_AVG16         = const(0xC)  # 8.51 ms
_ADC_SETTINGS_12BIT_AVG32         = const(0xD)  # 17.02 ms  ~= 1 60-Hz cycle
_ADC_SETTINGS_12BIT_AVG64         = const(0xE)  # 34.05 ms
_ADC_SETTINGS_12BIT_AVG128        = const(0xF)  # 68.10 ms

_CONFIG_MODE_POWERDOWN            = const(0x0)
_CONFIG_MODE_S_TRIGGERED          = const(0x1)
_CONFIG_MODE_B_TRIGGERED          = const(0x2)
_CONFIG_MODE_S_B_TRIGGERED        = const(0x3)
_CONFIG_MODE_ADC_OFF              = const(0x4)
_CONFIG_MODE_S_CONTINUOUS         = const(0x5)
_CONFIG_MODE_B_CONTINUOUS         = const(0x6)
_CONFIG_MODE_S_B_CONTINUOUS       = const(0x7)


def _to_signed(num):
  if num > 0x7FFF:
    num -= 0x10000
  return num


# pylint: disable=old-style-class
class INA219:
  """Driver for the INA219 current sensor."""

  def __init__(self, i2c, addr=0x40):
    self.i2c_device = I2CDevice(i2c=i2c, device_address=addr)
    self._current_lsb = 0.0
    self._cal_value = 0
    self.configure()

  def configure(self):
    # Max current = 2.048A, LSB = 2.048A / 2^15 = 62.5uA
    # Cal = trunc(0.04096 / (62.5uA * 5mOhm)) = 131072
    self._cal_value = 131072
    self._current_lsb = 62.5e-6

    self._write_register(_REG_CALIBRATION, self._cal_value)
    self._write_register(_REG_CONFIG,
                         _CONFIG_B_VOLTAGERANGE_16V |
                         _CONFIG_GAIN_1_40MV |
                         (_ADC_SETTINGS_12BIT_AVG32 << _B_ADC_OFFSET) |
                         (_ADC_SETTINGS_12BIT_AVG32 << _S_ADC_OFFSET) |
                         _CONFIG_MODE_S_B_CONTINUOUS)

  def _write_register(self, reg, value):
    with self.i2c_device as i2c:
      i2c.write(bytearray([reg, (value >> 8) & 0xFF, value & 0xFF]))

  def _read_register(self, reg):
    buf = bytearray(3)
    buf[0] = reg
    with self.i2c_device as i2c:
      i2c.write(buf, start=0, end=1, stop=False)
      i2c.readinto(buf, start=1, end=3, stop=True)
    return (buf[1] << 8) | (buf[2])

  def voltage(self):
    """The bus voltage (between V- and GND). LSB is 4mV."""
    return _to_signed(self._read_register(_REG_BUSVOLTAGE) >> 3) * 4e-3

  def current(self):
    """The current through the series resistor. From Voltage LSB of 10uV."""
    return self._current_lsb * _to_signed(self._read_register(_REG_CURRENT))

```

**Comment by kattni** [comment]
It looks like several of those constants aren't used, which may explain why not all of Table 5 was included. Have you tested these changes? Would you be interested in putting in a PR with the changes you suggest? Thanks!

**Comment by barbudor** [comment]
Hi @kattni , 
I've fixed the values to match the datasheet(as identified by user above) and added the missing ones.
I also added new properties to access the config register and its fields.
I would like to submit a PR for the above.
Best regards

**Comment by kattni** [comment]
@barbudor Thank you for submitting the PR! I'm going to request a review from our group.

**Comment by sommersoft** [comment]
Fixed by #9. Thanks again @barbudor!
